### ISSUE TITLE
Heroku can run out of memory unexpectedly

### ISSUE THREAD

**Issue by ahaselsteiner** [ahaselsteiner/viroconweb#146]
**I'm submitting a ...**
  - [x] bug report
  - [ ] feature request
  - [ ] code improvement request

## Expected behavior
**How should it work (with the bug fixed or the feature implemented)?**
Heroku should not run out of memory or at least inform the user about it.

**Screenshot**

## Actual behavior
**How does it currently work (with the bug causing problems or without the feature)?**
Heroku runs out of memory and then for example does not allow to open the user profile site anymore. It only shows a "server error 500 page".

**Screenshot**

## Steps to reproduce the problem (how to see the actual behavior)

  1. In one session do a lot of calculations (measurement file, fitting, contours)
  2. Watch in the heroku logs how the memory exceeds 500 MB
  3. Click on "user profile" and receive the "error 500" page

**Comment by ahaselsteiner** [comment]
This is possible already solved since we now use with open and with urlopen. Before we had to manually close the file and we did not always do this. This might have caused the RAM to fill up.

![image](https://user-images.githubusercontent.com/11452055/39710110-26abea36-521c-11e8-960e-d0f1e0c40a68.png)

**Comment by ahaselsteiner** [comment]
The problem is still apparent:

I did the following:
10 times:
* upload measurement file
* fit probabilistic model to it
* calculate a contour

When trying to perform the 10th fit the application crashed:
![image](https://user-images.githubusercontent.com/11452055/39986997-b53889d4-5763-11e8-9137-4661f595e65f.png)

This is the log file showing the memory exceeding, an error R14 and then timing out very likely because of that:
![image](https://user-images.githubusercontent.com/11452055/39986984-ab44ec88-5763-11e8-9a44-d0027c8d08b7.png)

**Comment by ahaselsteiner** [comment]
We solved this by restricting the number of maximum requests:
![image](https://user-images.githubusercontent.com/11452055/40007235-82c177f2-579c-11e8-8f68-a54e6dc3817e.png)

I did the 10 times data upload + fitting + contour to test this.

If I understand this correctly, after this number of maximum requests is reached the application restarts freeing up the memory. This causes a slight delay and can be seen in the heroku log:
![image](https://user-images.githubusercontent.com/11452055/40007168-5c72c2d6-579c-11e8-9c76-5dd441009786.png)

In the future we could try to increase the maximum number of requests. However, for now this is solved.
